Vungle/VungleRouter.m

```diff
@@ -323,18 +340,18 @@ - (void)requestBannerAdWithPlacementID:(NSString *)placementID
     }
 }
 
-- (BOOL)isAdAvailableForPlacementId:(NSString *)placementId
+- (BOOL)isAdAvailableForDelegate:(id<VungleRouterDelegate>)delegate
 {
-    return [[VungleSDK sharedSDK] isAdCachedForPlacementID:placementId];
+    return [[VungleSDK sharedSDK] isAdCachedForPlacementID:[delegate getPlacementID] adMarkup:[delegate getAdMarkup]];
 }
 
-- (BOOL)isBannerAdAvailableForPlacementId:(NSString *)placementId size:(CGSize)size
+- (BOOL)isBannerAdAvailableForPlacementId:(NSString *)placementId size:(CGSize)size delegate:(id<VungleRouterDelegate>)delegate
```

Comment:
This method header can be simplified to just the delegate. There are getter methods for the placement id and size in the delegate class.

Reply:
u should do it.